test/geometry_tests.cpp

```diff
@@ -12,14 +12,154 @@
 
 namespace geometry {
 
-TEST(BresenhamTests, SamePoint) {
-  // GIVEN two origin points
-  // WHEN computing the pixels to iterpolate
-  const auto pixels = bresenham_conversion(0, 0, 0, 0);
-
-  // THEN it will just be the origin point.
-  std::vector<std::vector<int>> expected = {{0, 0}};
-  EXPECT_THAT(pixels, ::testing::ElementsAreArray(expected));
+void printPixels(std::vector<std::vector<int>> pixels){
+	std::cout << "\n";
+	for (const auto &pixel:pixels ) {
+		std::cout << "(" << pixel[0] << ", " << pixel[1] << ")\n";		
+	}		
+	std::cout << "\n";
+}
+
+bool allXOrYCoorSame(int x_or_y, std::vector<std::vector<int>> pixels){
+	if (x_or_y == 0) { // Check X coordinates
+		x_or_y = 0;
+	} else { // Check Y coordinatesd
+		x_or_y = 1;
+	}
+
+	auto shouldBeEqualTo = pixels[0][x_or_y];
+
+	float sum = 0;
+
+	for (const auto &pixel:pixels) {
+		sum += pixel[x_or_y];
+	}
+
+	float avg = sum/(float)pixels.size();
```

Comment:
Change to `static_cast<float>(pixels.size())`

Reply:
Done